runner.vm/files/usr/local/sbin/network.sh

```diff
@@ -19,6 +19,12 @@ sysctl -q -w net.ipv6.conf.eth0.disable_ipv6=1
 ip addr add 169.254.0.0/16 dev eth0
 ip link set eth0 up
 
+tries=100
+until eval '(( $(< /sys/class/net/eth0/carrier) ))'; do
+   (( tries-- > 0 )) || exit 1
+   sleep 0.1
+done
+
```

Comment:
This loop only verifies that an Ethernet cable is connected...not that the network interface is actually functional. Here are two suggestions from ChatGPT to check that a network interface is functional in a BASH script, both of which I tried in my terminal.

The first starts a service dependent on `network-online.target` from the `NetworkManager-wait-online.service`. It waits until this target is reached, then returns.
```bash
systemd-run --wait --quiet --service-type=oneshot --property="After=network-online.target" /bin/true
```

This second example is simpler, it reads a file like the one that you have.
```
cat /sys/class/net/eth0/operstate
```
We would want this command to return `up`.

From ChatGPT:
> `/sys/class/net/eth0/operstate` is a sysfs file that shows the operational state of the network interface `eth0`. The possible values for this file are `unknown`, `notpresent`, `down`, `lowerlayerdown`, `testing`, `dormant`, `up`. The `operstate` file reflects the current state of the network interface based on various factors, including the physical state of the interface, the state of the underlying protocol stack, and the administrative state of the interface.
>
> `/sys/class/net/eth0/carrier` is another sysfs file that indicates the physical state of the network interface `eth0`. This file shows whether the link is up or down and is used to determine if the network cable is connected to the interface. The possible values for this file are `0` and `1`, where `0` indicates that the link is down and `1` indicates that the link is up.
>
> In summary, `operstate` reflects the overall state of the network interface, while `carrier` indicates the physical state of the link.